CMakeLists.txt

```diff
@@ -1660,6 +1660,10 @@ add_option("WOLFSSL_USER_SETTINGS"
     "Use your own user_settings.h and do not add Makefile CFLAGS (default: disabled)"
     "no" "yes;no")
 
+add_option("WOLFSSL_USER_SETTINGS_ASM"
+    "Enable use of user_settings_asm.h in assembly files (default: disabled)"
```

Comment:
Still in the process of building and testing this change, but I think we should default to generating the user_settings_asm.h file unless set not to when WOLFSSL_USER_SETTINGS is enabled. 

Something along the lines override_cache() call to explicetly turn of the asm file if the an option is set to disable it when WOLFSSL_USER_SETTINGS is set.

Reply:
Even if that breaks the CMake build when shellscript isn't available? That's the issue this PR was trying to fix.